packages/dds/tree/api-report/tree.api.md

```diff
@@ -1705,6 +1713,9 @@ export function stackTreeFieldCursor<TNode>(adapter: CursorAdapter<TNode>, root:
 // @internal
 export function stackTreeNodeCursor<TNode>(adapter: CursorAdapter<TNode>, root: TNode): CursorWithNode<TNode>;
 
+// @public
+export type StoredKey = string;
```

Comment:
So far our pattern has been to avoid using type alias like this for our public APIs since they don't work well with type checking or refactoring tools. We have just been using the underlying types in carefully documented ways.

Internally we use branded types for additional type safety where needed.

Since this type appears very little in the public API, and internally aligns with our branded FieldKey type, I don't think exporting this as public here is desired.

Reply:
They don't provide type-safety, but they do offer a convenient _single_ place to document concepts like this. I am personally a big fan of this pattern, with or without added type-safety.

If this package is going out of its way to avoid this sort of pattern, I can comply here, but I think this adds value.

Interested in others' opinions on this as well.

Reply:
Note that in our API report, FieldProps.key does not use this type. Thus I think this type isn't really doing its job.

I do think having a central place for docs to link to is good: FieldProps.key might be a fine central place to document this. That is where all stable keys are declared.

Also I don't think the typing is quite a useful/robust as it might appear in this case. For example its not super obvious that in Tree.child you can provide a string thats a map key: that stableKey really only applies to objects.

Reply:
I noticed that I hadn't actually updated `FieldProps.key` to use it at all. That has now been updated.

I'll still keep this thread alive to see if others have strong opinions.

Reply:
I agree that it's a convenient, centralized place to put documentation. If there is no other way we can document it consistently, it seems like it is warranted. On the whole though, I don't like type aliases very much. There are ways that they don't do what you expect, e.g. mousing-over in VS Code in certain scenarios shows the original type, not the alias, and they don't show up in type errors consistently. And it's just one more thing for the user to have to ctrl+click and hold in their brain ("StoredKey means string, okay, remember that...").

Reply:
Alright. I think I'll go ahead and remove this type alias then. It also occurred to me that after various refactorings, the `FieldProps` are the only remaining place where the concept appears in the public API. So using that as the centralized documentation seems reasonable.